programs/test/udp_proxy.c

```diff
@@ -613,11 +616,16 @@ int main( int argc, char *argv[] )
     int nb_fds;
     fd_set read_fds;
 
+    mbedtls_platform_context platform_ctx;
+
+    mbedtls_platform_setup( &platform_ctx );
```

Comment:
I think we're making the code unnecessarily more complex by placing part of the initialisation before the call to `get_options()`. Because of that, `get_options()` and `exit_usage()` now have to provide the same exit path to ensure `mbedtls_platform_teardown()` is called.

In my view, we should analyse and parse the options *before* we begin platform initialisation. What purpose is there to initialising the platform and network before we know whether we can execute?

I think `get_options()` should be called before `mbedtls_platform_setup()` and `mbedtls_net_init()`. If you do that you also didn't have to rewrite the options parsing and usage, which I'd revert.

Reply:
In principle I assumed that the platform initialization may potentially impact any standard function, therefore if the users expect the platform specific code to be always enabled for the given platform the initialization function should be called as soon as possible.

For example if the platform initialization impacts the file system access and the options parsing involves opening files, it will depend on the platform initialization having been called beforehand.

Please let me know if I'm mistaken. I will start reverting relevant changes immediately (this has been done for more than one example), but maybe it is a valid approach after all?